notebooks/bedrock/Financial Multilingual Embeddings.ipynb

```diff
@@ -0,0 +1,545 @@
+   "id": "ab9e4080-542b-4aad-94f4-6cf33789c792",
+   "metadata": {},
+   "source": [
+    "### Step 2: Import Documents \n",
+    "\n",
+    "Information about MultiFIN can be found in its Github repo https://github.com/RasmusKaer/MultiFin. The repo contains a link to [Dropbox](https://www.dropbox.com/sh/v0fvtn5gnf4ij6f/AABPSV4NzHIRa8GwUfQ7kA0aa?dl=0) where the data can be downloaded.\n",
+    "\n",
+    "Download the file _MultiFinDataset2023.zip_, unzip the file and navigate into the folder _MultiFinDataset_EACL_ and then _0-all-languages-lowlevel_. It contains a _train.json_ file. Upload that file to your Jupyter SageMaker Notebook."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "94c04e40-e8ca-4273-b4df-d5b8e01990d1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Read the data\n",
+    "df = pd.read_json(\"./train.json\", lines=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7197a0b7-f01e-46bd-ad2d-49395d119cd2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Inspect dataset\n",
+    "df.head(5)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "327177dd-bc70-4259-b3d4-4b7a70eedd1e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# check language distribution\n",
+    "df['lang'].value_counts()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0facfeb0-92e9-4709-b69f-1f2f60db7c43",
+   "metadata": {},
+   "source": [
+    "### Step 3. Select List of Documents to Query"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "fcbafe7d-db5b-4f3d-8234-0ec577761cdd",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Ensure there is no duplicated text in the headers\n",
+    "def remove_duplicates(text):\n",
+    "    return re.sub(r'((\\b\\w+\\b.{1,2}\\w+\\b)+).+\\1', r'\\1', text, flags=re.I)\n",
+    "\n",
+    "df ['text'] = df['text'].apply(remove_duplicates)\n",
+    "\n",
+    "# Select languages\n",
+    "languages = ['English', 'Spanish', 'Danish']\n",
+    "top_80_df = df.loc[df['lang'].isin(languages)]\n",
+    "\n",
+    "# Pick the top 80 longest articles\n",
+    "top_80_df['text_length'] = top_80_df['text'].str.len()\n",
+    "top_80_df.sort_values(by=['text_length'], ascending=False, inplace=True)\n",
+    "top_80_df = top_80_df[:80]\n",
+    "\n",
+    "# Language distribution\n",
+    "top_80_df['lang'].value_counts()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "09fb0837-91c2-4a5d-8f9a-9772690036f7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Let's look at the first record\n",
+    "top_80_df['text'].iloc[0]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9aa02a01",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Set up translator. You can also commemnt this block and use the Python list at the end of this notebook\n",
```

Comment:
I ran into translation issues and it was a bit of a pain to debug.
What if we just provided the train.json file (to avoid the Dropbox link going down) and included the translated titles in the file to begin with? Then people won't have to worry about this translation stuff.

Reply:
I agree that providing the train.json is better. I don't have translations for the thousands of files though. Let's discuss on Monday.

Reply:
csv file with translations uploaded in commit 3521452eab4a05147372c6449c641e3dc257040d

Reply:
Now the code points to the dataset in the github repo, you just gotta change the branch from 'main' to 'financial-multilingual-embed-notebook' when testing the notebook before merging the request

Reply:
thank you!